File: Data Structures & Algorithms/maximum-subarray/submission-0.java
class Solution {
    public int maxSubArray(int[] nums) {

        int currentSum = 0;
        int maxSub = nums[0];

        for(int num : nums) {
            
            currentSum += num;
            maxSub = Math.max(currentSum, maxSub);
            
            if (currentSum < 0) {
                currentSum = 0;
            }
        }

        return maxSub;
        
    }
}

File: Data Structures & Algorithms/max-water-container/submission-0.cs
public class Solution {
    public int MaxArea(int[] heights) {

        int max = 0;
        int l = 0;
        int r = heights.Length - 1;

        while (l < r) {

            int top = Math.Min(heights[l], heights[r]);
            max = Math.Max(max, (r - l) * top);

            if (heights[r] > heights[l]) {
                l++;
            } else {
                r--;
            }

        }

        return max;
        
    }
}
